cmake.Modules/CompileFlags.cmake

```diff
@@ -24,7 +24,7 @@ IF(UNIX)
         SET(CMAKE_MACOSX_RPATH 1)
     ENDIF()
 
-    SET (CXX_OPTIONS       " -std=c++11 -Wno-deprecated -Wall -Werror -Wshadow -Wextra -Wfloat-equal -Wpointer-arith -Wwrite-strings -Wpacked")
+    SET (CXX_OPTIONS       " -std=c++11 -Wno-deprecated -Wall -Wextra -Wfloat-equal -Wpointer-arith -Wwrite-strings -Wpacked")

```

Comment:
Sorry for the delay.

So I tried building the repo with -Werror and -Wshadow. I didn't change anything else and made sure that I'm up to date with master. I got a lot of errors because of -Wshadow and fixed them all. Finally this is what I get:

> [  2%] Building CXX object CMakeFiles/CommonLib.dir/src/common/utils.cpp.o
[  4%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/simulationframedata.cpp.o
[  6%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/recordingsection.cpp.o
[  8%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/dirtyframe.cpp.o
[ 10%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/feature.cpp.o
[ 12%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/permutation.cpp.o
[ 14%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/featureset.cpp.o
In file included from /home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/dirtyframe.h:1:0,
                 from /home/sm/simulation-model-matching/src/common/data/featureset.h:6,
                 from /home/sm/simulation-model-matching/src/common/data/featureset.cpp:1:
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/dirtyframe.h: In constructor ‘Common::DirtyFrame::DirtyFrame()’:
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/dirtyframe.h:13:5: warning: ‘Common::DirtyFrame::features’ should be initialized in the member initialization list [-Weffc++]
     DirtyFrame() = default;
     ^
/home/sm/simulation-model-matching/src/common/data/featureset.cpp: In constructor ‘Common::FeatureSet::FeatureSet(const string&, const string&)’:
/home/sm/simulation-model-matching/src/common/data/featureset.cpp:6:69: note: synthesized method ‘Common::DirtyFrame::DirtyFrame()’ first required here 
     : recordingName(_recordingName), id(_id), emptyFrame(DirtyFrame()) {
                                                                     ^
/home/sm/simulation-model-matching/src/common/data/featureset.cpp:5:1: warning: ‘Common::FeatureSet::frames’ should be initialized in the member initialization list [-Weffc++]
 FeatureSet::FeatureSet(const std::string& _recordingName, const std::string& _id)
 ^
[ 17%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/region.cpp.o
[ 19%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/rectangle.cpp.o
[ 21%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/coordinates.cpp.o
[ 23%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/simulationdata.cpp.o
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp: In constructor ‘Common::SimulationData::SimulationData()’:
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp:9:5: warning: ‘Common::SimulationData::simulationName’ should be initialized in the member initialization list [-Weffc++]
     SimulationData::SimulationData() {
     ^
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp:9:5: warning: ‘Common::SimulationData::correlationFile’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp:9:5: warning: ‘Common::SimulationData::id’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp:9:5: warning: ‘Common::SimulationData::permutationIndex’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp:9:5: warning: ‘Common::SimulationData::computationTimes’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp:9:5: warning: ‘Common::SimulationData::memory’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/common/data/simulationdata.cpp:9:5: warning: ‘Common::SimulationData::steeringWheelAngles’ should be initialized in the member initialization list [-Weffc++]
[ 25%] Building CXX object CMakeFiles/CommonLib.dir/src/common/data/correlation.cpp.o
/home/sm/simulation-model-matching/src/common/data/correlation.cpp: In constructor ‘Common::Correlation::Correlation(const Common::Rectangle&)’:
/home/sm/simulation-model-matching/src/common/data/correlation.cpp:5:5: warning: ‘Common::Correlation::sections’ should be initialized in the member initialization list [-Weffc++]
     Correlation::Correlation(const Rectangle& _rectangle)
     ^
[ 27%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/featuresetdao.cpp.o
[ 29%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/database.cpp.o
[ 31%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/correlationdao.cpp.o
[ 34%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/ronidao.cpp.o
[ 36%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/tablecolumn.cpp.o
[ 38%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/dataaccessobject.cpp.o
[ 40%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/tablefield.cpp.o
[ 42%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/cursor.cpp.o
[ 44%] Building CXX object CMakeFiles/CommonLib.dir/src/common/datamanagement/simulationdatadao.cpp.o
[ 46%] Linking CXX static library libCommonLib.a
[ 46%] Built target CommonLib
[ 48%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/odvcontrol.cpp.o
/home/sm/simulation-model-matching/src/simulationgame/odvcontrol.cpp: In constructor ‘SimulationGame::OdvControl::OdvControl()’:
/home/sm/simulation-model-matching/src/simulationgame/odvcontrol.cpp:10:5: warning: ‘SimulationGame::OdvControl::runningProcesses’ should be initialized in the member initialization list [-Weffc++]
     OdvControl::OdvControl() {
     ^
[ 51%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/datagatherer.cpp.o
/home/sm/simulation-model-matching/src/simulationgame/datagatherer.cpp: In constructor ‘SimulationGame::DataGatherer::DataGatherer()’:
/home/sm/simulation-model-matching/src/simulationgame/datagatherer.cpp:10:5: warning: ‘SimulationGame::DataGatherer::data’ should be initialized in the member initialization list [-Weffc++]
     DataGatherer::DataGatherer()
     ^
/home/sm/simulation-model-matching/src/simulationgame/datagatherer.cpp:10:5: warning: ‘SimulationGame::DataGatherer::frameStartTime’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/datagatherer.cpp:10:5: warning: ‘SimulationGame::DataGatherer::frameTime’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/datagatherer.cpp:10:5: warning: ‘SimulationGame::DataGatherer::dao’ should be initialized in the member initialization list [-Weffc++]
[ 53%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/abstractautopilot.cpp.o
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp: In constructor ‘SimulationGame::AbstractAutopilot::AbstractAutopilot(int, char**, const string&)’:
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp:16:5: warning: ‘SimulationGame::AbstractAutopilot::showGui’ should be initialized in the member initialization list [-Weffc++]
     AbstractAutopilot::AbstractAutopilot(int argc, char** argv, const std::string& name)
     ^
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp:16:5: warning: ‘SimulationGame::AbstractAutopilot::localFeatureSets’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp:16:5: warning: ‘SimulationGame::AbstractAutopilot::lastSegment’ should be initialized in the member initialization list [-Weffc++]
In file included from /home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/rectangle.h:1:0,
                 from /home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.h:16,
                 from /home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp:1:
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/rectangle.h: In constructor ‘constexpr Common::Rectangle::Rectangle()’:
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/rectangle.h:12:9: warning: ‘Common::Rectangle::topLeft’ should be initialized in the member initialization list [-Weffc++]
         Rectangle() = default;
         ^
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/rectangle.h:12:9: warning: ‘Common::Rectangle::bottomRight’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp: In constructor ‘SimulationGame::AbstractAutopilot::AbstractAutopilot(int, char**, const string&)’:
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp:17:85: note: synthesized method ‘constexpr Common::Rectangle::Rectangle()’ first required here 
         : odcore::base::module::TimeTriggeredConferenceClientModule(argc, argv, name) {
                                                                                     ^
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp:16:5: warning: ‘SimulationGame::AbstractAutopilot::dataGatherer’ should be initialized in the member initialization list [-Weffc++]
     AbstractAutopilot::AbstractAutopilot(int argc, char** argv, const std::string& name)
     ^
/home/sm/simulation-model-matching/src/simulationgame/abstractautopilot.cpp:16:5: warning: ‘SimulationGame::AbstractAutopilot::vehicleControl’ should be initialized in the member initialization list [-Weffc++]
[ 55%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/settings.cpp.o
[ 57%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/gamerunner.cpp.o
/home/sm/simulation-model-matching/src/simulationgame/gamerunner.cpp: In constructor ‘SimulationGame::GameRunner::GameRunner(SimulationGame::AutopilotRegistry&)’:
/home/sm/simulation-model-matching/src/simulationgame/gamerunner.cpp:12:5: warning: ‘SimulationGame::GameRunner::control’ should be initialized in the member initialization list [-Weffc++]
     GameRunner::GameRunner(SimulationGame::AutopilotRegistry& _registry)
     ^
/home/sm/simulation-model-matching/src/simulationgame/gamerunner.cpp:12:5: warning: ‘SimulationGame::GameRunner::correlationDao’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/gamerunner.cpp:12:5: warning: ‘SimulationGame::GameRunner::featureSetDao’ should be initialized in the member initialization list [-Weffc++]
In file included from /home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/permutation.h:1:0,
                 from /home/sm/simulation-model-matching/src/simulationgame/gamerunner.h:3,
                 from /home/sm/simulation-model-matching/src/simulationgame/gamerunner.cpp:1:
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/permutation.h: In constructor ‘Common::Permutation::Permutation()’:
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/permutation.h:14:9: warning: ‘Common::Permutation::index’ should be initialized in the member initialization list [-Weffc++]
         Permutation() = default;
         ^
/home/sm/simulation-model-matching/include/FeatureSimulation/Common/Data/../../../../src/common/data/permutation.h:14:9: warning: ‘Common::Permutation::localFeatureSets’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/gamerunner.cpp: In member function ‘std::vector<Common::Permutation> SimulationGame::GameRunner::createEmptyPermutation() const’:
/home/sm/simulation-model-matching/src/simulationgame/gamerunner.cpp:82:46: note: synthesized method ‘Common::Permutation::Permutation()’ first required here 
         result.push_back(Common::Permutation());
                                              ^
[ 59%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/simsettingsreader.cpp.o
In file included from /home/sm/simulation-model-matching/include/FeatureSimulation/SimulationGame/settings.h:1:0,
                 from /home/sm/simulation-model-matching/src/simulationgame/simsettingsreader.h:5,
                 from /home/sm/simulation-model-matching/src/simulationgame/simsettingsreader.cpp:1:
/home/sm/simulation-model-matching/include/FeatureSimulation/SimulationGame/../../../src/simulationgame/settings.h: In constructor ‘SimulationGame::Settings::Settings()’:
/home/sm/simulation-model-matching/include/FeatureSimulation/SimulationGame/../../../src/simulationgame/settings.h:7:12: warning: ‘SimulationGame::Settings::configurationFile’ should be initialized in the member initialization list [-Weffc++]
     struct Settings {
            ^
/home/sm/simulation-model-matching/include/FeatureSimulation/SimulationGame/../../../src/simulationgame/settings.h:7:12: warning: ‘SimulationGame::Settings::id’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/include/FeatureSimulation/SimulationGame/../../../src/simulationgame/settings.h:7:12: warning: ‘SimulationGame::Settings::autopilot’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/include/FeatureSimulation/SimulationGame/../../../src/simulationgame/settings.h:7:12: warning: ‘SimulationGame::Settings::correlationFile’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/simsettingsreader.cpp: In static member function ‘static SimulationGame::Settings SimulationGame::SimSettingsReader::readSettings(int, char**)’:
/home/sm/simulation-model-matching/src/simulationgame/simsettingsreader.cpp:53:18: note: synthesized method ‘SimulationGame::Settings::Settings()’ first required here 
         Settings settings;
                  ^
[ 61%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/autopilotregistry.cpp.o
/home/sm/simulation-model-matching/src/simulationgame/autopilotregistry.cpp: In constructor ‘SimulationGame::AutopilotRegistry::AutopilotRegistry(const string&, const string&)’:
/home/sm/simulation-model-matching/src/simulationgame/autopilotregistry.cpp:5:5: warning: ‘SimulationGame::AutopilotRegistry::autopilots’ should be initialized in the member initialization list [-Weffc++]
     AutopilotRegistry::AutopilotRegistry(const std::string& _cid, const std::string& _freq)
     ^
[ 63%] Building CXX object CMakeFiles/SimulationGameLib.dir/src/simulationgame/autopilots/lanefollower.cpp.o
/home/sm/simulation-model-matching/src/simulationgame/autopilots/lanefollower.cpp: In constructor ‘SimulationGame::LaneFollower::LaneFollower(int, char**)’:
/home/sm/simulation-model-matching/src/simulationgame/autopilots/lanefollower.cpp:7:5: warning: ‘SimulationGame::LaneFollower::m_previousTime’ should be initialized in the member initialization list [-Weffc++]
     LaneFollower::LaneFollower(int argc, char** argv)
     ^
/home/sm/simulation-model-matching/src/simulationgame/autopilots/lanefollower.cpp:7:5: warning: ‘SimulationGame::LaneFollower::m_eSum’ should be initialized in the member initialization list [-Weffc++]
/home/sm/simulation-model-matching/src/simulationgame/autopilots/lanefollower.cpp:7:5: warning: ‘SimulationGame::LaneFollower::m_eOld’ should be initialized in the member initialization list [-Weffc++]
[ 65%] Linking CXX static library libSimulationGameLib.a
[ 65%] Built target SimulationGameLib
[ 68%] Building CXX object CMakeFiles/simulationGame.dir/src/main_simulationgame.cpp.o
[ 70%] Linking CXX executable simulationGame
[ 70%] Built target simulationGame
[ 72%] Building CXX object CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/settingsvalidator.cpp.o
[ 74%] Building CXX object CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/settings.cpp.o
[ 76%] Building CXX object CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/featuredetectororb.cpp.o
[ 78%] Building CXX object CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/guicontroler.cpp.o
/home/sm/simulation-model-matching/src/featuresetcreation/guicontroler.cpp: In constructor ‘FeatureSetCreation::GuiController::GuiController()’:
/home/sm/simulation-model-matching/src/featuresetcreation/guicontroler.cpp:8:5: warning: ‘FeatureSetCreation::GuiController::image’ should be initialized in the member initialization list [-Weffc++]
     GuiController::GuiController() {
     ^
[ 80%] Building CXX object CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/featuredetector.cpp.o
[ 82%] Building CXX object CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/recordingplayer.cpp.o
In file included from /home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:1:0:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:25: error: ‘SharedMemory’ is not a member of ‘odcore::wrapper’
         std::shared_ptr<odcore::wrapper::SharedMemory> extractSharedMemoryFromContainer();
                         ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:25: error: ‘SharedMemory’ is not a member of ‘odcore::wrapper’
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:54: error: template argument 1 is invalid
         std::shared_ptr<odcore::wrapper::SharedMemory> extractSharedMemoryFromContainer();
                                                      ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp: In constructor ‘FeatureSetCreation::RecordingPlayer::RecordingPlayer(std::__cxx11::string)’:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:14:5: warning: ‘FeatureSetCreation::RecordingPlayer::currentContainer’ should be initialized in the member initialization list [-Weffc++]
     RecordingPlayer::RecordingPlayer(std::string recordingFile)
     ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp: In member function ‘cv::Mat FeatureSetCreation::RecordingPlayer::next()’:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:52:101: error: conversion from ‘int’ to non-scalar type ‘std::shared_ptr<odcore::wrapper::SharedMemory>’ requested
             std::shared_ptr<odcore::wrapper::SharedMemory> memory = extractSharedMemoryFromContainer();
                                                                                                     ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:53:82: error: invalid use of incomplete type ‘class odcore::wrapper::SharedMemory’
             return cv::Mat(imageSize(), CV_MAKETYPE(CV_8U, bytesPerPixel), memory->getSharedMemory()).clone();
                                                                                  ^
In file included from /home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:7:0:
/usr/include/opendavinci/odcore/wrapper/SharedMemoryFactory.h:31:7: note: forward declaration of ‘class odcore::wrapper::SharedMemory’
 class SharedMemory;
       ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp: At global scope:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:59:52: error: prototype for ‘std::shared_ptr<odcore::wrapper::SharedMemory> FeatureSetCreation::RecordingPlayer::extractSharedMemoryFromContainer()’ does not match any in class ‘FeatureSetCreation::RecordingPlayer’
     std::shared_ptr<odcore::wrapper::SharedMemory> RecordingPlayer::extractSharedMemoryFromContainer() {
                                                    ^
In file included from /home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:1:0:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:56: error: candidate is: int FeatureSetCreation::RecordingPlayer::extractSharedMemoryFromContainer()
         std::shared_ptr<odcore::wrapper::SharedMemory> extractSharedMemoryFromContainer();
                                                        ^
CMakeFiles/FeatureSetCreationLib.dir/build.make:182: recipe for target 'CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/recordingplayer.cpp.o' failed
make[2]: *** [CMakeFiles/FeatureSetCreationLib.dir/src/featuresetcreation/recordingplayer.cpp.o] Error 1
CMakeFiles/Makefile2:142: recipe for target 'CMakeFiles/FeatureSetCreationLib.dir/all' failed
make[1]: *** [CMakeFiles/FeatureSetCreationLib.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2



As you can see, my code still produces some warnings about member initialization wherever I use the default implementation of a constructor. I can fix those, if you like, but I don't think they matter.

The problem I can't fix is this

> In file included from /home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:1:0:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:25: error: ‘SharedMemory’ is not a member of ‘odcore::wrapper’
         std::shared_ptr<odcore::wrapper::SharedMemory> extractSharedMemoryFromContainer();
                         ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:25: error: ‘SharedMemory’ is not a member of ‘odcore::wrapper’
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:54: error: template argument 1 is invalid
         std::shared_ptr<odcore::wrapper::SharedMemory> extractSharedMemoryFromContainer();
                                                      ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp: In constructor ‘FeatureSetCreation::RecordingPlayer::RecordingPlayer(std::__cxx11::string)’:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:14:5: warning: ‘FeatureSetCreation::RecordingPlayer::currentContainer’ should be initialized in the member initialization list [-Weffc++]
     RecordingPlayer::RecordingPlayer(std::string recordingFile)
     ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp: In member function ‘cv::Mat FeatureSetCreation::RecordingPlayer::next()’:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:52:101: error: conversion from ‘int’ to non-scalar type ‘std::shared_ptr<odcore::wrapper::SharedMemory>’ requested
             std::shared_ptr<odcore::wrapper::SharedMemory> memory = extractSharedMemoryFromContainer();
                                                                                                     ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:53:82: error: invalid use of incomplete type ‘class odcore::wrapper::SharedMemory’
             return cv::Mat(imageSize(), CV_MAKETYPE(CV_8U, bytesPerPixel), memory->getSharedMemory()).clone();
                                                                                  ^
In file included from /home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:7:0:
/usr/include/opendavinci/odcore/wrapper/SharedMemoryFactory.h:31:7: note: forward declaration of ‘class odcore::wrapper::SharedMemory’
 class SharedMemory;
       ^
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp: At global scope:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:59:52: error: prototype for ‘std::shared_ptr<odcore::wrapper::SharedMemory> FeatureSetCreation::RecordingPlayer::extractSharedMemoryFromContainer()’ does not match any in class ‘FeatureSetCreation::RecordingPlayer’
     std::shared_ptr<odcore::wrapper::SharedMemory> RecordingPlayer::extractSharedMemoryFromContainer() {
                                                    ^
In file included from /home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.cpp:1:0:
/home/sm/simulation-model-matching/src/featuresetcreation/recordingplayer.h:21:56: error: candidate is: int FeatureSetCreation::RecordingPlayer::extractSharedMemoryFromContainer()
         std::shared_ptr<odcore::wrapper::SharedMemory> extractSharedMemoryFromContainer();


I don't know what happened to SharedMemory.